airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/engage.py

```diff
@@ -165,6 +166,7 @@ def process_response(self, response: requests.Response, stream_state: Mapping[st
             if not item_cursor or not state_cursor or item_cursor >= state_cursor:
                 yield item
 
+    @cache
```

Comment:
Why do we need to cache the schema? It should be available right after the `discover` is finished inside catalog.json, correct me if i'm wrong.

Reply:
it's streams with dynamically generated schema, just to avoid executing `get_json_schema` twice